src/App.tsx

```diff
@@ -8,26 +8,27 @@ const projectRegex = /^P[a-zA-Z0-9]{27}$/;
 const App = () => {
 	const [error, setError] = useState(false);
 
-	const baseUrl = process.env.REACT_APP_DESCOPE_BASE_URL || '';
+	const baseUrl =
+		process.env.REACT_APP_DESCOPE_BASE_URL || window.location.origin;
 
 	let projectId = '';
 
 	// first, take project id from env
-	const envProjectId = process.env.DESCOPE_PROJECT_ID;
-	if (envProjectId && projectRegex.test(envProjectId)) {
-		projectId = envProjectId;
-	}
+	const envProjectId = projectRegex.exec(
```

Comment:
Why exec and not test? 
Also, where is the envProjectId used?

Reply:
exec so it returns the captured group
moved the selection of projectId to https://github.com/descope/auth-hosting/pull/42/files#diff-26ad4b834941d9b19ebf9db8082bd202aaf72ea0ddea85f5a8a0cb3c729cc6f2R30 because it's only relevant after navigating away from the `/login/project_id` part